WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListFragment.kt

```diff
@@ -352,11 +447,35 @@ class ProductListFragment :
                     event.selectedCategoryName
                 )
                 is ShowProductSortingBottomSheet -> showProductSortingBottomSheet()
+                is SelectProducts -> tracker?.setItemsSelected(event.productsIds, true)
                 else -> event.isHandled = false
             }
         }
     }
 
+    private fun handleListState(productListState: ProductListViewModel.ProductListState) {
+        when (productListState) {
+            ProductListViewModel.ProductListState.Selecting -> {
+                delayMultiSelection()
+                onListSelectionActiveChanged(true)
+                enableProductsRefresh(false)
+                enableProductSortAndFiltersCard(false)
+            }
+            ProductListViewModel.ProductListState.Browsing -> {
+                onListSelectionActiveChanged(false)
```

Comment:
⚙️ We should also clear selection when entering browsing state. Otherwise, the state is inconsistent.

```suggestion
            ProductListViewModel.ProductListState.Browsing -> {
                tracker?.clearSelection()
                onListSelectionActiveChanged(false)
```